src/basekernels/wiener.jl

```diff
@@ -47,16 +47,16 @@ function WienerKernel(;i::Integer=0)
 end
 
 function (::WienerKernel{0})(x, y)
-    X = sqrt(sum(abs2, x))
-    Y = sqrt(sum(abs2, y))
-    return min(X, Y)
+    X_2 = sum(abs2, x)
+    Y_2 = sum(abs2, y)
+    return sqrt(min(X_2, Y_2))
 end
 
 function (::WienerKernel{1})(x, y)
-    X = sqrt(sum(abs2, x))
-    Y = sqrt(sum(abs2, y))
-    minXY = min(X, Y)
-    return 1 / 3 * minXY^3 + 1 / 2 * minXY^2 * euclidean(x, y)
+    X_2 = sum(abs2, x)
+    Y_2 = sum(abs2, y)
+    minX2Y2 = min(X_2, Y_2)
+    return 1 / 3 * minX2Y2^(3/2) + 1 / 2 * minX2Y2 * euclidean(x, y)
```

Comment:
Good opportunity to get rid of any unintended promotions as well:
```suggestion
    return minX2Y2^(3/2) / 3 + minX2Y2 * euclidean(x, y) / 2
```

Reply:
BTW I just benchmarked `f(x) = sqrt(x)^3` and `g(x) = x^(3/2)`, and the first one is much faster (due to some compiler optimizations I guess) so you definitely want to use
```suggestion
    return sqrt(minX2Y2)^3 + minX2Y2 * euclidean(x, y) / 2
```